src/main/hover/goHover.ts

```diff
@@ -28,4 +32,87 @@ export class GoHover extends AbstractHoverProvider {
         }
         return undefined;
     }
+
+    provideHover(document: vscode.TextDocument, position: vscode.Position): vscode.ProviderResult<vscode.Hover> {
```

Comment:
There is a lot of code repetition in this function. Inside it and between it and its parent. Please fix!